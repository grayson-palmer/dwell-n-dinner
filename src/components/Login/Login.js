import React, { Component } from 'react';
import './Login.scss';
import { Link } from 'react-router-dom';

export default class Login extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      email: '',
      reason: ''
    };
  }

  handleChange = event => {
    this.setState({ [event.target.name]: event.target.value });
  };

  loginUser = e => {
    e.preventDefault();
    const { setUserInfo } = this.props;
    setUserInfo({ ...this.state });
  };

  render() {
    return (
      <form className="login-form">
        <h1>Dwell-n-Dine</h1>
        <label htmlFor="login-name">Name</label>
        <input
          onChange={this.handleChange}
          name="name"
          id="login-name"
          placeholder="Name"
        ></input>
        <label htmlFor="login-email">Email</label>
        <input
          onChange={this.handleChange}
          name="email"
          id="login-email"
          placeholder="Email"
        ></input>
        <label htmlFor="login-reason">Reason for visit</label>
        <select onChange={this.handleChange} id="login-reason" name="reason">
          <option disabled selected value>
            -- select an option --
          </option>
          <option value="Special Occasion">Special Occasion</option>
          <option value="Business">Business</option>
          <option value="Vacation">Vacation</option>
          <option value="Other">Other</option>
        </select>
        <Link className="login-button" to="/Areas">
          Login
        </Link>
      </form>
    );
  }
}
