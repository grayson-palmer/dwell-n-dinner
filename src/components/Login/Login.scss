@import '../../styles/mixins.scss';
@import '../../styles/variables';

.login-form {
@include text-bg-colors;
  border: 0.75rem double $main-text-color;
  border-radius: 1rem;
  display: flex;
  flex-direction: column;
  font-size: 1.5em;
  justify-content: space-around;
  padding: 2em;
  width: 40%;
  > label {
    margin: 0 0 0.25em;
    &:not(:first-of-type) {
      margin-top: 1em;
    }
  }
  > input, select {
    border-radius: 0.25em;
    font-size: 1em;
    padding: 0.25em;
  }
  > h1 {
    margin: 0 0 0.5em;
  }
}

.login-button {
  @include button-styles;
}
