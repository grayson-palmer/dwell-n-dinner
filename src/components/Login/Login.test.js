import React from 'react';
import { shallow } from 'enzyme';
import Login from './Login';

describe('Login', () => {
  it('', () => {
    
  })
})