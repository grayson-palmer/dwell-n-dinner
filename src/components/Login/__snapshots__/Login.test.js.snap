// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render the component 1`] = `
<form
  className="login-form"
>
  <h1>
    Dwell-n-Dine
  </h1>
  <label
    htmlFor="login-name"
  >
    Name
  </label>
  <input
    id="login-name"
    name="name"
    onChange={[Function]}
    placeholder="Name"
  />
  <label
    htmlFor="login-email"
  >
    Email
  </label>
  <input
    id="login-email"
    name="email"
    onChange={[Function]}
    placeholder="Email"
  />
  <label
    htmlFor="login-reason"
  >
    Reason for visit
  </label>
  <select
    id="login-reason"
    name="reason"
    onChange={[Function]}
  >
    <option
      disabled={true}
      selected={true}
      value={true}
    >
      -- select an option --
    </option>
    <option
      value="Special Occasion"
    >
      Special Occasion
    </option>
    <option
      value="Business"
    >
      Business
    </option>
    <option
      value="Vacation"
    >
      Vacation
    </option>
    <option
      value="Other"
    >
      Other
    </option>
  </select>
  <Link
    className="login-button"
    to="/areas"
  >
    Login
  </Link>
</form>
`;
